Xamarin.Forms.Platform.Android/FastRenderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,465 @@
+			MediaElement = (MediaElement)element;
+
+			Performance.Start(out string reference);
+
+			if (oldElement != null)
+			{
+				oldElement.PropertyChanged -= OnElementPropertyChanged;
+			}
+
+			Color currentColor = oldElement?.BackgroundColor ?? Color.Default;
+			if (element.BackgroundColor != currentColor)
+			{
+				UpdateBackgroundColor();
+			}
+
+			MediaElement.PropertyChanged += OnElementPropertyChanged;
+			MediaElement.SeekRequested += SeekRequested;
+			MediaElement.StateRequested += StateRequested;
+
+			if (_tracker is null)
+			{
+				// Can't set up the tracker in the constructor because it access the Element (for now)
+				SetTracker(new VisualElementTracker(this));
+			}
+
+			OnElementChanged(new ElementChangedEventArgs<MediaElement>(oldElement as MediaElement, MediaElement));
+
+			EffectUtilities.RegisterEffectControlProvider(this, oldElement, element);
+
+			Performance.Stop(reference);
+		}
+
+		void StateRequested(object sender, StateRequested e)
+		{
+			switch (e.State)
+			{
+				case MediaElementState.Playing:
+					_view.Start();
+					Controller.CurrentState = _view.IsPlaying ? MediaElementState.Playing : MediaElementState.Stopped;
+					break;
+
+				case MediaElementState.Paused:
+					if (_view.CanPause())
+					{
+						_view.Pause();
+						Controller.CurrentState = MediaElementState.Paused;
+					}
+					break;
+
+				case MediaElementState.Stopped:
+					_view.Pause();
+					_view.SeekTo(0);
+
+					Controller.CurrentState = _view.IsPlaying ? MediaElementState.Playing : MediaElementState.Stopped;
+					break;
+			}
+
+			UpdateLayoutParameters();
+			Controller.Position = _view.Position;
+		}
+
+		void SeekRequested(object sender, SeekRequested e)
+		{
+			Controller.Position = _view.Position;
+		}
+
+		void IVisualElementRenderer.SetLabelFor(int? id)
+		{
+			if (_defaultLabelFor is null)
+			{
+				_defaultLabelFor = LabelFor;
+			}
+
+			LabelFor = (int)(id ?? _defaultLabelFor);
+		}
+		void SetTracker(VisualElementTracker tracker)
+		{
+			_tracker = tracker;
+		}
+
+		void UpdateBackgroundColor()
+		{
+			SetBackgroundColor(Element.BackgroundColor.ToAndroid());
+		}
+
+		void IVisualElementRenderer.UpdateLayout() => _tracker?.UpdateLayout();
+
+
+		protected override void Dispose(bool disposing)
+		{
+			if (_isDisposed)
+			{
+				return;
+			}
+
+			_isDisposed = true;
+
+			if (disposing)
+			{
+				SetOnClickListener(null);
+				SetOnTouchListener(null);
+
+				_automationPropertiesProvider?.Dispose();
+				_tracker?.Dispose();
+
+				if (Element != null)
+				{
+					Element.PropertyChanged -= OnElementPropertyChanged;
+
+					if (Platform.GetRenderer(Element) == this)
+						Element.ClearValue(Platform.RendererProperty);
+				}
+			}
+
+			base.Dispose(disposing);
+		}
+
+		void OnElementChanged(ElementChangedEventArgs<MediaElement> e)
+		{
+			if (e.OldElement != null)
+			{
+				ReleaseControl();
+			}
+
+			if (e.NewElement != null && !(_view is null))
+			{
```

Comment:
This is the only one block where the `_view` value is set. And to get here you need to have the value already been set...

Reply:
I've changed this as the view can live for the duration of the renderer. I've moved the code to set it up into the constructor and moved the cleanup code into Dispose. Changing the Element then merely needs to reset the properties based on the new element.